internals/controllers/subnamespace_controller.go

```diff
@@ -177,15 +205,22 @@ func (r *SubnamespaceReconciler) Sync(ownerNamespace *utils.ObjectContext, subsp
 		free[res] = *resource.NewQuantity(value, resource.BinarySI)
 	}
 
-	if subspaceparent.IsPresent() {
-		r.SnsEvents <- event.GenericEvent{Object: &danav1.Subnamespace{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      subspaceparent.Object.GetName(),
-				Namespace: subspaceparent.Object.GetNamespace(),
-			},
-		}}
+	for _, sns := range subspaceChilds.Objects.(*danav1.SubnamespaceList).Items {
+		snsChild, err := utils.NewObjectContext(subspace.Ctx, subspace.Log, subspace.Client, types.NamespacedName{Name: sns.GetName(), Namespace: subspace.GetName()}, &danav1.Subnamespace{})
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		if utils.GetSnsResourcePooled(subspace.Object) == "false" &&
+			utils.GetSnsResourcePooled(snsChild.Object) == "true" &&
+			snsChild.Object.GetAnnotations()[danav1.IsUpperRp] == danav1.False {
```

Comment:
This `if` statement is way to convoluted, move it to a function with a relevant name that returns a boolean; will make this more readable as well